### PR TITLE
LPS-167659 Generic error for duplicate ERC field on Picklist

### PR DESCRIPTION
**Pull request by daiazius**
Related issue: https://issues.liferay.com/browse/LPS-167659

### CODE REVIEW

**Comment by daiazius** [comment]
ci:test:sf

**Comment by daiazius** [comment]
ci:test:relevant

File: modules/apps/list-type/list-type-service/src/main/java/com/liferay/list/type/service/impl/ListTypeDefinitionLocalServiceImpl.java [hunk 1]
```diff
@@ -149,6 +149,17 @@ public ListTypeDefinition updateListTypeDefinition(
...
+			int count = listTypeDefinitionPersistence.countByC_ERC(
+				listTypeDefinition.getCompanyId(), externalReferenceCode);
+
+			if (count != 0) {
+				throw new DuplicateListTypeExternalReferenceCodeException(
+					"This External Reference Code is already in use. Try " +
+						"another one.");
+			}
+		}
+
```
[the remark this answers is not in the document]
Comment (samuelromeiro) [reply modules/apps/list-type/list-type-service/src/main/java/com/liferay/list/type/service/impl/ListTypeDefinitionLocalServiceImpl.java:162]: Hey @daiazius, can you check Mateus' comments, please?

[the remark this answers is not in the document]
Comment (daiazius) [reply modules/apps/list-type/list-type-service/src/main/java/com/liferay/list/type/service/impl/ListTypeDefinitionLocalServiceImpl.java:162]: Hey @samuelromeiro, I'm working on it

[the remark this answers is not in the document]
Comment (murilostodolni) [reply modules/apps/list-type/list-type-service/src/main/java/com/liferay/list/type/service/impl/ListTypeDefinitionLocalServiceImpl.java:162]: Hi @samuelromeiro I'm helping @daiazius with this. We will have a problem with this approach...

**Comment by daiazius** [comment]
ci:test:sf

**Comment by daiazius** [comment]
ci:test:sf

File: modules/apps/list-type/list-type-service/src/main/java/com/liferay/list/type/service/persistence/impl/ListTypeDefinitionPersistenceImpl.java [hunk 2]
```diff
@@ -2630,8 +2630,7 @@ public ListTypeDefinition updateImpl(
 			if (isNew) {
 				if (ercListTypeDefinition != null) {
 					throw new DuplicateListTypeDefinitionExternalReferenceCodeException(
-						"Duplicate list type definition with external reference code " +
-							listTypeDefinition.getExternalReferenceCode());
+						"This External Reference Code is already in use. Try another one.");
```
Comment (murilostodolni) [inline modules/apps/list-type/list-type-service/src/main/java/com/liferay/list/type/service/persistence/impl/ListTypeDefinitionPersistenceImpl.java:2633]: Hi @daiazius there is some reason to change this message?

File: modules/apps/list-type/list-type-service/src/main/java/com/liferay/list/type/service/persistence/impl/ListTypeDefinitionPersistenceImpl.java [hunk 3]
```diff
@@ -2640,8 +2639,7 @@ public ListTypeDefinition updateImpl(
 						ercListTypeDefinition.getListTypeDefinitionId())) {
 
 					throw new DuplicateListTypeDefinitionExternalReferenceCodeException(
-						"Duplicate list type definition with external reference code " +
-							listTypeDefinition.getExternalReferenceCode());
+						"This External Reference Code is already in use. Try another one.");
```
Comment (murilostodolni) [inline modules/apps/list-type/list-type-service/src/main/java/com/liferay/list/type/service/persistence/impl/ListTypeDefinitionPersistenceImpl.java:2642]: @daiazius same above

**Review by murilostodolni — CHANGES_REQUESTED** [review]
Good job! Just a comment.

File: modules/apps/list-type/list-type-service/src/main/java/com/liferay/list/type/service/persistence/impl/ListTypeDefinitionPersistenceImpl.java (hunk 2 above)
Comment (daiazius) [reply modules/apps/list-type/list-type-service/src/main/java/com/liferay/list/type/service/persistence/impl/ListTypeDefinitionPersistenceImpl.java:2633]: I changed it to be the same as requested in the issue

**Review by murilostodolni — APPROVED** [review]
LGTM 🎉

**Comment by daiazius** [comment]
ci:test:sf

**Comment by daiazius** [comment]
ci:test:relevant

**Comment by daiazius** [comment]
ci:test:relevant

**Comment by murilostodolni** [comment]
ci:stop

**Comment by murilostodolni** [comment]
ci:stop

**Comment by murilostodolni** [comment]
Forwarding manually: https://github.com/brianchandotcom/liferay-portal/pull/126949

**Comment by murilostodolni** [comment]
ci:close

**Pull request closed without merging**